src/registry/domain/s3.js

```diff
@@ -148,13 +148,7 @@ module.exports = function(conf) {
 
           putFile(file, url, relativeFile === '/server.js', cb);
         },
-        errors => {
-          if (errors) {
-            return callback(_.compact(errors));
-          }
-
-          callback(null, 'ok');
-        }
+        callback
```

Comment:
async.each actually fires the error as soon as it gets the first. This means that the "compact" is not necessary as the error is just 1. When testing locally, I was seeing a error with a `[Object object]` result here prior of this fix in case of S3 issues.